File: ccc/2007-j2-txtmsg.py
# Declare and populate a dictionary of translations.
#
# A dictionary (a.k.a. associative array, map) is a python object
# (usually denoted by curly brackets) that lets you "look up" any
# object given another object very quickly. The lookup is very
# fast -- about log(n) time.
translation = {
    'CU': "see you",
    ':-)': "I'm happy",
    ':-(': "I'm unhappy",
    ';-)': "wink",
    ':-P': "stick out my tongue",
    '(~.~)': "sleepy",
    'TA': "totally awesome",
    'CCC': "Canadian Computing Competition",
    'CUZ': "because",
    'TY': "thank-you",
    'YW': "you're welcome",
    'TTYL': "talk to you later"
}

# Python doesn't have do-while, so we use an infinite loop that we
# break out of when the user enters 'TTYL'.
while True:
    word = input('Enter phrase> ')

    # This is a concise way to write
    #
    # if word in translation:
    #     print(translation[word])
    # else:
    #     print(word)
    print(translation[word] if word in translation else word)

    if word == 'TTYL': break

File: ccc/2007-j2-txtmsg.cpp
#include <iostream>
#include <map>
using namespace std;

int main ()
{
    /* Declare a map from string to string.
     *
     * A map (a.k.a. associative array, dictionary) is a data structure
     * where you can "look up" any object given another object. The lookup is
     * usually very fast -- about log(n) time. */
    map<string, string> translation;

    string input;

    /* Populate our map with the given translations. */
    translation["CU"] = "see you";
    translation[":-)"] = "I'm happy";
    translation[":-("] = "I'm unhappy";
    translation[";-)"] = "wink";
    translation[":-P"] = "stick out my tongue";
    translation["(~.~)"] = "sleepy";
    translation["TA"] = "totally awesome";
    translation["CCC"] = "Canadian Computing Competition";
    translation["CUZ"] = "because";
    translation["TY"] = "thank-you";
    translation["YW"] = "you're welcome";
    translation["TTYL"] = "talk to you later";

    /* Continue to get input and either translate it,
     * or echo it (if there is no translation) ... as long as
     * the input is not "TTYL". */
    do {
        cout << "Enter phrase> ";
        cin >> input;

        if (translation.count(input))
            cout << translation[input] << endl;
        else 
            cout << input << endl;
    } while (input != "TTYL");

    // That is all.
    return 0;
}
